core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala

```diff
@@ -101,8 +101,8 @@ private[spark] class MetricsSystem private (
     running = true
     if (registerStaticSources) {
       StaticSources.allSources.foreach(registerSource)
-      registerSources()
     }
+    registerSources()
```

Comment:
This is a partial revert of another your previous commit, https://github.com/apache/spark/pull/22279, two year ago, isn't it?
```
[SPARK-25277][YARN] YARN applicationMaster metrics should not register static metrics
```

Reply:
What happens in YARN applicationMaster after this PR?